File: test1.c
#include<stdio.h>
#include<string.h>
#include <sys/time.h>
#include <math.h>
#include <stdlib.h>

int myadd(int a, int b) {
    return a+b;
}

int cmpfunc( const void *a, const void *b) {
  return *(char*)a - *(char*)b;
}

int fib(int a) {
    if (a == 0 || a == 1)
        return a;
    else
        return fib(a-1) + fib(a-2);
}

int main(){
    int i,j,count;
    double sum = 0;
    double add = 1;
    long seconds = 0;
    long microseconds = 0;
    double elapsed = 0;
    struct timeval begin, end;

    gettimeofday(&begin, 0);
    for (i=0; i< 10000000; i++) {        
    }
    gettimeofday(&end, 0);
    seconds = end.tv_sec - begin.tv_sec;
    microseconds = end.tv_usec - begin.tv_usec;
    elapsed = seconds + microseconds*1e-6;    
    printf("10m loop: %.3f seconds.\n", elapsed);

    gettimeofday(&begin, 0);
    for (i=0; i< 1000000; i++) {        
       j = 1/pow(i+1,0.5);
    }
    gettimeofday(&end, 0);
    seconds = end.tv_sec - begin.tv_sec;
    microseconds = end.tv_usec - begin.tv_usec;
    elapsed = seconds + microseconds*1e-6;    
    printf("1m pow loop: %.3f seconds.\n", elapsed);

    char * buffer = 0;
    long length;
    gettimeofday(&begin, 0);
    FILE * f = fopen ("tale-of-two-cities64x.txt", "rb");
    fseek (f, 0, SEEK_END);
    length = ftell (f);
    fseek (f, 0, SEEK_SET);
    buffer = malloc (length);
    if (buffer)
    {
        fread (buffer, 1, length, f);
    }
    gettimeofday(&end, 0);
    fclose (f);
    seconds = end.tv_sec - begin.tv_sec;
    microseconds = end.tv_usec - begin.tv_usec;
    elapsed = seconds + microseconds*1e-6;    
    printf("read 11m word file: %.3f seconds.\n", elapsed);

    buffer = 0;
    f = fopen ("tale-of-two-cities.txt", "rb");
    fseek (f, 0, SEEK_END);
    length = ftell (f);
    fseek (f, 0, SEEK_SET);
    buffer = malloc (length);
    if (buffer)
    {
        fread (buffer, 1, length, f);
    }
    gettimeofday(&begin, 0);
    qsort(buffer, (size_t) strlen(buffer), (size_t) sizeof(char), cmpfunc);
    gettimeofday(&end, 0);
    fclose (f);
    seconds = end.tv_sec - begin.tv_sec;
    microseconds = end.tv_usec - begin.tv_usec;
    elapsed = seconds + microseconds*1e-6;    
    printf("sort 792k words: %.3f seconds.\n", elapsed);

    gettimeofday(&begin, 0);
    fib(40);
    gettimeofday(&end, 0);
    seconds = end.tv_sec - begin.tv_sec;
    microseconds = end.tv_usec - begin.tv_usec;
    elapsed = seconds + microseconds*1e-6;    
    printf("fib 40: %.3f seconds.\n", elapsed);

    gettimeofday(&begin, 0);
    for (i=0; i< 10000000; i++) {       
        myadd(i,myadd(i,1)); 
    }
    gettimeofday(&end, 0);
    seconds = end.tv_sec - begin.tv_sec;
    microseconds = end.tv_usec - begin.tv_usec;
    elapsed = seconds + microseconds*1e-6;    
    printf("10m add-func loop: %.3f seconds.\n", elapsed);

    gettimeofday(&begin, 0);
    for (i=0; i< 10000000; i++) {       
        i+i+1; 
    }
    gettimeofday(&end, 0);
    seconds = end.tv_sec - begin.tv_sec;
    microseconds = end.tv_usec - begin.tv_usec;
    elapsed = seconds + microseconds*1e-6;    
    printf("10m add-inline loop: %.3f seconds.\n", elapsed);

    return 0;
}

File: test1.rs
use std::fs;
use std::net::SocketAddr;
use std::error::Error;

fn fib(a:i64) -> i64 {
    if a == 0 {
        return 0;
    }
    if a == 1 {
        return 1;
    }
    return fib(a-1) + fib(a-2);
}

fn add( a: i64, b:i64) -> i64 {
    return a+b;
}
    

fn main() {
    use std::time::Instant;
    let now = Instant::now();
    for n in 1..10000000 {    }
    let elapsed = now.elapsed();
    println!("loop 10m: {:.2?}", elapsed);

    let now = Instant::now();
    for n in 1..1000000 {
        1.0/(n as f64).sqrt();
    }
    let elapsed = now.elapsed();
    println!("sqrt 1m: {:.2?}", elapsed);

    let mut mystr = "".to_string();
    let now = Instant::now();
    for n in 1i64..500000 {
        mystr.push_str(&n.to_string());   
    }
    let elapsed = now.elapsed();
    println!("500k strcat: {:.2?}", elapsed);

    
    let mut mystr = "".to_string();
    let now = Instant::now();
    let book = fs::read_to_string("tale-of-two-cities64x.txt").expect("Something went wrong reading the file");
    let elapsed = now.elapsed();
    println!("11m file read: {:.2?}", elapsed);

    let book = fs::read_to_string("tale-of-two-cities.txt").expect("Something went wrong reading the file");
    let now = Instant::now();
    let mut l: Vec<char> = book.chars().collect();
    l.sort_unstable();
    let j: String = l.into_iter().collect();
    let elapsed = now.elapsed();
    println!("sort file: {:.2?}", elapsed);

    let now = Instant::now();
    fib(40);
    let elapsed = now.elapsed();
    println!("fib 40: {:.2?}", elapsed);
    
    let now = Instant::now();
    for n in 1..10000000 {
        add(n,add(n,1));
    }
    let elapsed = now.elapsed();
    println!("add 10m: {:.2?}", elapsed);

    let mut sum = 0;
    let now = Instant::now();
    for n in 1..10000000 {
        sum = n+n+1;
    }
    let elapsed = now.elapsed();
    println!("nofunc 10m: {:.2?}", elapsed);
}



